src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs

```diff
@@ -365,10 +360,8 @@ internal void SetThreadPoolWorkerThreadName()
 
             lock (this)
             {
-                // Bypass the exception from setting the property
                 _name = ThreadPool.WorkerThreadName;
                 ThreadNameChanged(ThreadPool.WorkerThreadName);
-                _name = null;
```

Comment:
_name was set to null here so that if the user workload tried to set the thread name, it would succeed. Thus below if _name was null, it was known the thread was "really" still WorkerThreadName, and otherwise then it should be reset. Now it is simple: _name doesn't lie.

Reply:
this also answers my question above

> not sure why, since the user code ought not to have been able to rename it successfully